Ft/integrate users' managements

Favor-star opened the pull request:
# Summary

This pull request introduces several improvements and refactors to the user management and authentication flows, focusing on dynamic data fetching, robust server-side pagination, and enhanced user experience for inviting and registering users. The registration and invitation processes now handle API-driven data, and the user list is dynamically loaded and paginated from the server.

---

## Type of Change

- [ ] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (may cause existing functionality to fail)
- [ ] UI/UX improvement
- [ ] Documentation update
- [x] Other (please specify): Integration of real data

---

## Implementation Details


**User Management Improvements**

* The `UsersList` component now fetches user data from the server using the `useFetchUsers` hook, supports server-side pagination, and displays loading/error states. This replaces the previous static mock data approach. (`app/dashboard/user-management/components/users-list.tsx`, `app/dashboard/user-management/page.tsx`, [[1]](diffhunk://#diff-114dcd59a28af8aba599232c71e2372ee301916f5ac339dd61d1cc3e317c48deR1-R49) [[2]](diffhunk://#diff-25e17b26142f9ca7aad2da15fd134e22549bc4d1de3bf022ed0389b7de25080cL8) [[3]](diffhunk://#diff-25e17b26142f9ca7aad2da15fd134e22549bc4d1de3bf022ed0389b7de25080cL32-R31)
* The `DataTable` component was refactored to support server-side pagination, including new props for pagination state and metadata, and improved display of user details (e.g., using `firstName`/`lastName`, formatting join date). (`components/custom/data-table.tsx`, [[1]](diffhunk://#diff-82863d7c88b0e0968451fbe46a8303bc0b712798dd90dec7889c4b724d3d81e2R30) [[2]](diffhunk://#diff-82863d7c88b0e0968451fbe46a8303bc0b712798dd90dec7889c4b724d3d81e2L68-R69) [[3]](diffhunk://#diff-82863d7c88b0e0968451fbe46a8303bc0b712798dd90dec7889c4b724d3d81e2L97-L109) [[4]](diffhunk://#diff-82863d7c88b0e0968451fbe46a8303bc0b712798dd90dec7889c4b724d3d81e2L193-R247)

**Invite User Flow Enhancements**

* The `InviteUserForm` now fetches available roles from the API, displays loading/error states, and updates the user list after a successful invite. It also supports closing the form automatically. (`app/dashboard/user-management/components/invite-user-form.tsx`, [[1]](diffhunk://#diff-1902763ad5eb6cbd78a04531091077bd6301e054e31f7be6804f01c60bc92b0eL8-R27) [[2]](diffhunk://#diff-1902763ad5eb6cbd78a04531091077bd6301e054e31f7be6804f01c60bc92b0eL27-R82) [[3]](diffhunk://#diff-1902763ad5eb6cbd78a04531091077bd6301e054e31f7be6804f01c60bc92b0eL49-R105)
* The `RadioGroupComponent` now dynamically renders role options based on fetched data rather than hardcoded values. (`app/dashboard/user-management/components/radio-group.tsx`, [[1]](diffhunk://#diff-d5b9076a000ffd78b695d65019b28574160285749bc4c9fec1202c9119f22264R11-R16) [[2]](diffhunk://#diff-d5b9076a000ffd78b695d65019b28574160285749bc4c9fec1202c9119f22264L24-R35)

**Registration Flow Updates**

* The registration form now requires invite tokens and pre-fills the email field, enforcing registration only via invite. It also adds username input and disables editing of the email field. Success and error handling are improved with toast notifications and redirects. (`app/(auth)/register/components/register-form.tsx`, [[1]](diffhunk://#diff-bba6651e0f0d3dadc25de3f8ae5f105610b2c63ac5990e2e1ef44dc1e0cedbceR11-R53) [[2]](diffhunk://#diff-bba6651e0f0d3dadc25de3f8ae5f105610b2c63ac5990e2e1ef44dc1e0cedbceL43-R104)
* The registration page layout was slightly adjusted for improved spacing. (`app/(auth)/register/page.tsx`, [app/(auth)/register/page.tsxL6-R6](diffhunk://#diff-bdc4811d55ad8e0c687a98535b7be088f01df1f5728d8181733a14787e1dc04aL6-R6))

**Minor UI/UX Changes**

* The login form no longer displays the "Sign Up" link, reflecting the invite-only registration flow. (`app/(auth)/login/components/login-form.tsx`, [app/(auth)/login/components/login-form.tsxL111-L121](diffhunk://#diff-5712121dfc7bb1f0c5d17b742731b4cdcacd16c1c098a8084a7419ede274a393L111-L121))
* Pagination logic in the `Pagination` component was cleaned up for clarity. (`components/custom/pagination.tsx`, [components/custom/pagination.tsxL35](diffhunk://#diff-4458db649f4ac8b21ad71d9690f529e76c86fb16be6facb6f13022d23881c9c9L35))


---

## How Has This Been Tested?

Explain how you tested your changes. Provide steps for reviewers to reproduce:  
- [ ] Unit tests  
- [ ] Integration tests  
- [x] Manual testing  
- [x] Visual/UI testing  

---

## Screenshots / Recordings (if applicable)

Attach before/after screenshots, screen recordings, or GIFs to visualize UI changes.

---

## Related Changes

- [ ] Routes updated  
- [ ] API contract changes  
- [ ] Config/environment variables changed  
- [ ] Dependency upgrades  

---

## Checklist

- [x] My code follows project style guidelines
- [x] I have self-reviewed my code
- [ ] I have commented complex parts of the code
- [ ] Documentation has been updated (if needed)
- [x] No new warnings or errors
- [ ] Tests added or updated
- [x] All tests pass locally
- [ ] Relevant issues linked and closed

---

Favor-star committed:
feat: fetch roles and stores them in the redux store

Favor-star committed:
feat: intergrates user fetching and populated data table, and intergate form for inviting users

Favor-star committed:
fix: fixed pagination to implement manual pagination on the users' table

Favor-star committed:
feat: implemented complete registration with the provided token, and removed link to register on login(since non-invited users will not have access to register)

Favor-star committed:
fix: changed type of ListData and styled the status badge  and changed its types to use enum

Favor-star committed:
feat: integrated single user page with user info, waiting to integrate the edit user form

Bob-Karemera-Shema commented on app/(auth)/register/components/register-form.tsx:
```diff
@@ -8,52 +8,100 @@ import Link from "next/link";
-      type: "manual",
-      message: "Unable to register, please try again. ",
+    const inviteToken = searchParams.get("invite") ?? "";
+    const registerResponse = await completeUserRegister(inviteToken, {
```
Would it be possible to implement this in a cleaner way?

Bob-Karemera-Shema reviewed (commented):
Overall everything looks good, just double check the comment I left to see if something can be done about it.

Chartine02 commented on app/dashboard/user-management/components/invite-user-form.tsx:
```diff
@@ -24,15 +33,53 @@ export const InviteUserForm = () => {
+        action: {
+          label: "Undo",
+          onClick: () => {
+            // Handle undo action
```
Shouldn't you implement this too ?

Favor-star replied:
This logic would require intervention of the backend because I would need to also delete the data in the database. However, as things stands now, such implementation is not yet done from the backend side. So, that is why I left that function empty.

Favor-star committed:
fix: fixed string errors on the register user schema on completring user registration

Chartine02 commented on app/dashboard/user-management/components/users-list.tsx:
```diff
@@ -1,23 +1,72 @@
+        />
+        No users found
+      </section>
+    );
```
I would suggest wrapping the logic inside the `section` to avoid repetition

Chartine02 commented on lib/api/users.ts:
```diff
@@ -0,0 +1,83 @@
+
+    return await response.json();
+  } catch (error) {
+    console.error("Error inviting user:", error);
```
Since you have return the error and handling it accordingly.
I would suggest removing the console.

Chartine02 reviewed (changes requested):
Nice work, you can check on the requested changes.

Favor-star committed:
fix(pr-review): removal of console.error, fixing repetition, and restructuring the invite user form

Favor-star committed:
refactor(sidebar): include a user info in the sidebar info card

Favor-star committed:
refactor(singleuser page): add a sidebar trigger when the sidebar have collapsed

feat(auth): included  the api-request for server, and refactored fetching requests with API

Bob-Karemera-Shema commented on lib/types/api.ts:
```diff
@@ -31,12 +41,14 @@ export interface PageInfo {
-  totalPages: number;
-  hasNext: boolean;
-  hasPrevious: boolean;
+  pagination: {
```
Consider using PageInfo type
